sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/condition_node.py

```diff
@@ -63,16 +69,37 @@ def _validate_params(self, raise_error=True) -> MutableValidationResult:
                     f"with value 'True', got {output_definition.is_control}",
                 )
 
-        error_msg = "{!r} of dsl.condition node must be an instance of " f"{BaseNode} or {AutoMLJob}," "got {!r}."
-        if self.true_block is not None and not isinstance(self.true_block, (BaseNode, AutoMLJob)):
+        # check if condition is valid binding
+        if isinstance(self.condition, str) and not is_data_binding_expression(
+                self.condition, ["parent"], is_singular=False):
+            error_tail = "for example, ${{parent.jobs.xxx.outputs.output}}"
+            validation_result.append_error(
+                yaml_path="condition",
+                message=f"'condition' of dsl.condition has invalid binding expression: {self.condition}, {error_tail}",
+            )
+
+        error_msg = "{!r} of dsl.condition node must be an instance of " \
+                    f"{BaseNode}, {AutoMLJob} or {str}," "got {!r}."
+        if self.true_block is not None and not isinstance(self.true_block, (BaseNode, AutoMLJob, str)):
             validation_result.append_error(
                 yaml_path="true_block", message=error_msg.format("true_block", type(self.true_block))
             )
-        if self.false_block is not None and not isinstance(self.false_block, (BaseNode, AutoMLJob)):
+        if self.false_block is not None and not isinstance(self.false_block, (BaseNode, AutoMLJob, str)):
             validation_result.append_error(
                 yaml_path="false_block", message=error_msg.format("false_block", type(self.false_block))
             )
 
+        # check if true/false block is valid binding
+        for name, block in {"true_block": self.true_block, "false_block": self.false_block}.items():
+            if block is None or not isinstance(block, str):
```

Comment:
Can we put the above lines in this for? As the above lines are both built on 'name' and 'block'.